src/main/scala/mesosphere/marathon/core/launchqueue/impl/ReviveOffersStreamLogic.scala

```diff
@@ -136,13 +58,109 @@ object ReviveOffersStreamLogic extends StrictLogging {
     */
   def suppressAndReviveFlow(
     minReviveOffersInterval: FiniteDuration,
-    enableSuppress: Boolean): Flow[Either[InstanceChangeOrSnapshot, DelayedStatus], Op, NotUsed] = {
+    enableSuppress: Boolean): Flow[Either[InstanceChangeOrSnapshot, DelayedStatus], RoleDirective, NotUsed] = {
+
+    val reviveRepeaterWithTicks = Flow[RoleDirective]
+      .map(Left(_))
+      .merge(Source.tick(minReviveOffersInterval, minReviveOffersInterval, Right(Tick)))
```

Comment:
Really 🙂 I had the tick in an earlier draft and we rejected it again. What is the purpose of the tick? Aren't we throttling anyways?

Reply:
Heh, yeah, it was difficult to do without implementing a custom graph stage. The logic for whether or not we repeat is based off of whether or not that role is suppressed or not.

I'm not super happy about it, either. It works. :/